superset-frontend/packages/superset-ui-chart-controls/src/shared-controls/constants.tsx

```diff
@@ -22,45 +22,17 @@ import {
   t,
   validateNonEmpty,
 } from '@superset-ui/core';
-import { ControlPanelState, ControlState } from '../types';
+import { ControlPanelState } from '../types';
 
-export const xAxisControlConfig = {
-  label: (state: ControlPanelState) => {
-    if (
-      isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES) &&
-      state?.form_data?.orientation === 'horizontal'
-    ) {
-      return t('Y-axis');
-    }
+const getAxisLabel = (state: ControlPanelState): [string, string] =>
+  isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES) &&
+  state?.form_data?.orientation === 'horizontal'
+    ? [t('Y-axis'), t('Dimension to use on y-axis.')]
+    : [t('X-axis'), t('Dimension to use on x-axis.')];
 
-    return t('X-axis');
-  },
-  default: (
-    control: ControlState,
-    controlPanel: Partial<ControlPanelState>,
-  ) => {
-    // default to the chosen time column if x-axis is unset and the
-    // GENERIC_CHART_AXES feature flag is enabled
-    const { value } = control;
-    if (value) {
-      return value;
-    }
-    const timeColumn = controlPanel?.form_data?.granularity_sqla;
-    if (isFeatureEnabled(FeatureFlag.GENERIC_CHART_AXES) && timeColumn) {
-      return timeColumn;
-    }
-    return null;
-  },
```

Comment:
IIRC, this logic was added so that charts that were created when the `GENERIC_CHART_AXES` FF was unset would render correctly when exploring them _after_ the GENERIC_CHART_AXES` FF was enabled. Otherwise it would open Explor and clear the chart due to the required `x_axis` control being unset. So the flow should be something like this:
- When entering Explore with FF disabled, nothing should happen (`x_axis` should be `null`)
- When entering Explore with the FF enabled and the `x_axis` value is unset, the value should be set to the temporal column
- When entering Explore with the FF enabled and the `x_axis` value is set, it should not be changed

Reply:
The X-Axis control `value` should be set instead of setting the `default` value because if the default value had been set, the X-Axis control wouldn't have chance to remove it(the X-AXIS always sets a default value). Let me find a way to set the X-Axis as the `sqla_grainularity` if `GENERIC_CHART_AXES` is enabled.

Reply:
Hi @villebro, I have added a new property for controls, `initialValue`. It refers to how to initialize a control when loading, this setting avoids the default value always set to the control.

I also tested charts between FF enabled and disabled.
1. Create a Bar Chart V2 when `GENERIC_CHART_AXES` is disabled, and then create a dashboard and take the chart.
2. Open the `GENERIC_CHART_AXES`
3. the Chart on the Explore page and Dashboard same as the disabled FF.

Reply:
> Hi @villebro, I have added a new property for controls, `initialValue`. It refers to how to initialize a control when loading, this setting avoids the default value always set to the control.

This is a really useful addition 👍 I'm sure this will be needed in the future at some point.